packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -255,30 +255,6 @@ export class CRNetworkManager {
       frame = this._page._frameManager.frameAttached(requestWillBeSentEvent.frameId, null);
     }
 
-    // CORS options preflight request is generated by the network stack. If interception is enabled,
-    // we accept all CORS options, assuming that this was intended when setting route.
-    //
-    // Note: it would be better to match the URL against interception patterns.
```

Comment:
Let's implement this url matching instead. That way you would fulfill all the analytics requests that you mentioned without going to the server while keeping the rest unaffected by the interception. And then all the tests (and the use cases they cover) that broke will keep working.